src/tests/class_tests/openms/source/CoarseIsotopeDistribution_test.cpp

```diff
@@ -143,6 +143,39 @@ START_SECTION(IsotopeDistribution convolve_(const CoarseIsotopePatternGenerator&
 }
 END_SECTION
 
+START_SECTION(( [EXTRA CH]IsotopeDistribution run(const EmpiricalFormula&) const ))
+{
+  EmpiricalFormula ef ("C6H12O6");
+
+  {
+    CoarseIsotopePatternGenerator gen(3);
+    IsotopeDistribution id = gen.run(ef);
+    TEST_EQUAL(id.size(), 3)
+
+    TEST_REAL_SIMILAR(id[0].getMZ(), 180.063)
+    TEST_REAL_SIMILAR(id[0].getIntensity(), 0.923456)
+
+    TEST_REAL_SIMILAR(id[2].getMZ(), 182.0701)
+    TEST_REAL_SIMILAR(id[2].getIntensity(), 0.013232)
+  }
+
+  // TODO: is that a good idea?
+  ef.setCharge(2);
```

Comment:
during testing I discovered that the `CoarseIsotopePatternGenerator` does not actually compute the isotopic envelope of the given sum formula but it uses `SumFormula + ch * H` if charge is set. This is i) unexpected and ii) assumes that the adduct is hydrogen. We should probably change that.

Reply:
I agree and close this subconversation because we have an issue for that